### PR TITLE
CMS Preview feature

### PR DESCRIPTION
**Pull request by margOnline**
[TP 9186](https://moneyadviceservice.tpondemand.com/restui/board.aspx?#page=board/5580782241563718174&appConfig=eyJhY2lkIjoiRTQ4MDEyNENDOUYxRDU4RDg0RTEyRjNCODBEN0VBMUYifQ==&searchPopup=userstory/9186)

## BEFORE MERGING
 - [x] Ensure [required changes to mas-cms-client](https://github.com/moneyadviceservice/mas-cms-client/pull/28) have been merged
 - [x] In the Gemfile, update the mas-cms-client gem version to match the version in the above pr

### Context
Fincap website editors need to preview their changes/additions to Fincap Evidence Summaries (Insights, Evaluations and Reviews),  once a draft version has been entered into the cms, before the document is published.

### Technical
- route to documents_preview_controller which despatches the request to
the preview controller corresponding to the page type
- the page_type preview controller gets the cms resource via mas-cms-client gem
- [required changes to mas-cms-client](https://github.com/moneyadviceservice/mas-cms-client/pull/28)

### CODE REVIEW

File: app/controllers/thematic_reviews_landing_pages_preview_controller.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class ThematicReviewsLandingPagesPreviewController < ThematicReviewsLandingPagesController
```
[the remark this answers is not in the document]
Comment (margOnline) [reply app/controllers/thematic_reviews_landing_pages_preview_controller.rb]: yes, the purpose of `thematic_reviews#index` shows summaries of all thematic_reviews plus some blurb at the top. The blurb is what the editors have entered into the cms for thematic_reviews_landing_page and the controller you refer to uses the `show` method to just preview that content.

File: app/controllers/thematic_reviews_landing_pages_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class ThematicReviewsLandingPagesController < FincapTemplatesController
```
[the remark this answers is not in the document]
Comment (margOnline) [reply app/controllers/thematic_reviews_landing_pages_controller.rb]: see comment below

File: config/initializers/mas_cms_client.rb [hunk 3]
```diff
@@ -19,3 +19,6 @@ class Mas::Cms::Review < Mas::Cms::Document
 
 class Mas::Cms::ThematicReview < Mas::Cms::Article
 end
+
+class Mas::Cms::ThematicReviewsLandingPage < Mas::Cms::Article
```
[the remark this answers is not in the document]
Comment (margOnline) [reply config/initializers/mas_cms_client.rb]: same comment

**Comment by margOnline** [comment]
@willhall88 
1. There is no authentication on preview pages on frontend. This is probably a wider query and outside the scope of this pr.
2. Preview pages should not be indexed. Search is not set up on this repo, perhaps a tp card is needed. Again probably outside the scope of this pr.
3. Have removed the ThematicReviewsLandingPage controller and allowed the corresponding preview_controller to inherit directly from FincapTemplatesController

**Comment by willhall88** [comment]
@margOnline 
1. agreed it's outside of this scope, just thought I would raise a question in case we need to discuss
2. I was talking more about indexing by crawlers i.e. google. we don't want the preview pages showing up in google search results for example. 

Will pull down everything shortly and test it properly

**Comment by masbuild** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          2 Errors
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><table>
  <tbody>
    <tr>
      <td>config/routes.rb</td>
      <td>10</td>
      <td>Style/SymbolArray: Use <code>%i</code> or <code>%I</code> for an array of symbols.</td>
    </tr>
  </tbody>
</table>
</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false"><table>
  <tbody>
    <tr>
      <td>config/routes.rb</td>
      <td>13</td>
      <td>Metrics/LineLength: Line is too long. [106/80]</td>
    </tr>
  </tbody>
</table>
</td>
    </tr>
  </tbody>
</table>

At Money Advice Service, we use Danger gem, to ensure that our
commits and pull requests follow  [standards](https://github.com/moneyadviceservice/mas-standards/blob/master/Dangerfile).
If you think one of the points here has been raised in error (or   shouldn't be in our standards), open an issue or pull request on [our   Danger config](https://github.com/moneyadviceservice/mas-standards).


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**